README.md

```diff
@@ -3,338 +3,64 @@ SMAUG: Simulating Machine Learning Accelerators Using gem5-Aladdin
+Users who cannot use Docker can follow the commands in the
+[Dockerfile](https://github.com/harvard-acc/smaug/blob/master/docker/Dockerfile)
+to set up a local environment. However, due to our limited resources, we can
+only offer support to Docker users.
 
-## Why does SMAUG exist? ##
+To install SMAUG, first install Docker, then pull the Docker image:
 
-The need for hardware acceleration of deep learning workloads is obvious, but
-understanding the impact of accelerator design choices on performance, power,
-and area often requires manual implementation of the accelerator, and writing
-RTL is slow. High level synthesis (HLS) can reduce this development time by
-allowing the developer to describe hardware in a higher level language, but
-ultimately, evaluation of the design requires the HLS to generate RTL, which
-can be a very time consuming process. Aladdin is a pre-RTL accelerator
-simulator which allows developers to explore the hardware design space by
-describing their accelerator designs in C, without actually needing to write
-RTL or wait for RTL to be generated.
+```bash
+docker pull xyzsam/smaug:latest
+```
 
-However, like HLS tools, Aladdin supports a subset of the C language, and the
-coding style required is rather unique and idiosyncratic. Writing C to describe
-hardware is radically different from writing C for high performance software.
-As a result, we cannot simply run code from existing optimized linear algebra
-libraries, like BLAS, MKL, or Eigen, through Aladdin.
+Then, run the following command to create a Docker volume that hosts your
+workspace and start the container. The local volume (aka `smaug-workspace`
+below) will store all your source code, local changes, and build artifacts, so
+that you can start/stop the Docker container without losing any of your work.
 
-SMAUG provides a set of implementation of core deep learning kernels that
-comply with Aladdin's coding style. Some of the implementations are merely
-reference implementations, while others represent actual accelerator designs.
-It also integrates these kernels with gem5-Aladdin, so that the combined SoC
-simulator can be used to measure overall end-to-end performance.  The user can
-decide which kernels to offload to specialized hardware and which kernels to
-run on the CPU.
+```bash
+docker run -it --rm --mount source=smaug-workspace,target=/workspace smaug
+```
 
-## Features ##
-Layer types supported:
-* Fully connected
-  - Arbitrary number of hidden nodes *
-* Convolutional
-  - Standard 2D convolution
-  - Depthwise separable convolution
-  - 1x1 convolution
-  - Arbitrary kernel size *
-* Pooling
-  - Average pooling
-  - Max pooling
+Once inside your Docker container, run the setup script to clone all the
+required source code repositories:
 
-Activation functions:
-* ReLU, ReLU clipped, leaky RELU
-* Sigmoid, tanh
-* ELU, SELU
-* Softmax
+```bash
+root@dockerimageid:/workspace # /setup_smaug_repos.sh
```

Comment:
I think the docker image already has all the repos, all the user needs to do is `git pull` and `git submodule update --init`.

I'm gonna approve this, we can add things in future commits.

Reply:
I updated this with the full on instructions. It's kind of ugly, so I think we're better off eventually adding a script to update the repos with one command.